bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/handler/ApiConsumerHandler.java

```diff
@@ -87,40 +87,62 @@ private void initializeOnceBridgeOnline(FreeboxOsHandler bridgeHandler) {
         Map<String, String> properties = editProperties();
         try {
             initializeProperties(properties);
-            checkAirMediaCapabilities(properties);
             updateProperties(properties);
         } catch (FreeboxException e) {
             logger.warn("Error getting thing {} properties: {}", thing.getUID(), e.getMessage());
         }
 
-        boolean isAudioReceiver = Boolean.parseBoolean(properties.get(MediaType.AUDIO.name()));
-        if (isAudioReceiver) {
-            configureMediaSink(bridgeHandler, properties.getOrDefault(Source.UPNP.name(), ""));
-        }
-
         startRefreshJob();
     }
 
-    private void configureMediaSink(FreeboxOsHandler bridgeHandler, String upnpName) {
+    protected void configureMediaSink() {
         try {
+            String upnpName = editProperties().getOrDefault(Source.UPNP.name(), "");
             Receiver receiver = getManager(MediaReceiverManager.class).getReceiver(upnpName);
-            if (receiver != null && reg == null) {
-                ApiConsumerConfiguration config = getConfig().as(ApiConsumerConfiguration.class);
-                String callbackURL = bridgeHandler.getCallbackURL();
-                if (!config.password.isEmpty() || !receiver.passwordProtected()) {
-                    reg = bridgeHandler.getBundleContext().registerService(
-                            AudioSink.class.getName(), new AirMediaSink(this, bridgeHandler.getAudioHTTPServer(),
-                                    callbackURL, receiver.name(), config.password, config.acceptAllMp3),
-                            new Hashtable<>());
-                } else {
-                    logger.info("A password needs to be configured to enable Air Media capability.");
-                }
+            if (receiver != null) {
+                Map<String, String> properties = editProperties();
+                receiver.capabilities().entrySet()
+                        .forEach(entry -> properties.put(entry.getKey().name(), entry.getValue().toString()));
+                updateProperties(properties);
+
+                startAudioSink(receiver);
+            } else {
+                stopAudioSink();
             }
         } catch (FreeboxException e) {
             logger.warn("Unable to retrieve Media Receivers: {}", e.getMessage());
         }
     }
 
+    private void startAudioSink(Receiver receiver) {
+        FreeboxOsHandler bridgeHandler = checkBridgeHandler();
+        // Only video and photo is supported by the API so use VIDEO capability for audio
+        Boolean isAudioReceiver = receiver.capabilities().get(MediaType.VIDEO);
+        if (reg == null && bridgeHandler != null && isAudioReceiver != null && isAudioReceiver.booleanValue()) {
+            ApiConsumerConfiguration config = getConfig().as(ApiConsumerConfiguration.class);
+            String callbackURL = bridgeHandler.getCallbackURL();
+            if (!config.password.isEmpty() || !receiver.passwordProtected()) {
```

Comment:
```suggestion
            if (!config.password.isBlank() || !receiver.passwordProtected()) {
```

Reply:
This is @clinique original code and it is correct. Your proposal would exclude a password containing only spaces. I agree this is very improbable and I see no reason to exclude it.

Reply:
For almost all password checks when isEmpty is used, isBlank is meant to be used. But ok. If we want to allow all whitespace passwords (yikes) I’ll merge this. 😁